src/AWS.Deploy.Orchestration/Utilities/DeployedApplicationQueryer.cs

```diff
@@ -137,5 +144,24 @@ public async Task<List<CloudApplication>> GetCompatibleApplications(List<Recomme
                 .OrderByDescending(x => x.LastUpdatedTime)
                 .ToList();
         }
+
+        /// <summary>
+        /// Gets the appropriate recommendation that will be used for a redeployment to the same Cloudformation stack. 
+        /// </summary>
+        public Recommendation? GetRecommendationForRedeployment(CloudApplication application, List<Recommendation> recommendations)
```

Comment:
I don't think this logic is going to work. Then GetRecommendationForRedeployment should return back a list or this method should take in a targetRecipeId. If a project currently has multiple saved deployment projects and the current application is using the parent recipe id. As I see this we are just taking whatever is the first match. Then up above we are comparing with the passed in deployment recipe id. And if that one wasn't the first one found here then we fail.

Reply:
Refer - https://github.com/aws/aws-dotnet-deploy/pull/318#discussion_r720445729